### ISSUE TITLE
Why not fine-tune the weights of conv layers in the experiment of ISIC

### ISSUE THREAD

**Issue by TreeLLi** [laura-rieger/deep-explanation-penalization#2]
Hi,

I notice that you choose to freeze the conv layers and only train the fully connected layers when using a pretrained VGG16 model in the ISIC experiment. Could you please, if convenient, tell me why not retrain the whole model since this seems to be an extra accuracy gain compared to training only fc layers.

All best.

**Comment by laura-rieger** [comment]
Hi, thanks for the question. We only finetune the last layers since the dataset, while huge for medical datasets, is relatively small by deep learning standards (~20.000 images compared to 10.000.000 for the original imagenet task). Retraining the entire network would overfit the dataset quickly.
Best, Laura

**Comment by TreeLLi** [comment]
thanks for your reply. I am wondering if you have experimented retraining the model and got an overfitting model. As shown in the paper, your result on RRR on ISIC is considerably worse than others and you thus conclude that "penalizing gradients is not helpful in penalizing higher-order features". (actually I am also confused by the term higher-order features, does this mean the features from the deeper model?) Have you ever considered the possibility that RRR regularisation fails due to the lack of control on the conv layers.

**Comment by laura-rieger** [comment]
We have tried this for the unregularized model and the model did indeed overfit. 

Higher order features are  not intermittent features in the network but features compromised by multiple pixels (f.e. the bandaids). I think it is unlikely that RRR fails due to lack of control on the conv layers: As the performance of the network shows, the features extracted from the conv layers are meaningful for classification. Thus, if RRR was able to regularize higher order features, it should be able to regularize usage of those features accordingly. 

The general consensus is also that the earlier conv layers extract basic image features (such as edges and corners), which is why transfer learning normally works quite well and freezing of the lower layers is recommended.
